src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -2282,9 +2282,9 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
         {
             assert(sig->numArgs == 2);
 
-            if (varTypeIsByte(simdBaseType))
+            if (varTypeIsByte(simdBaseType) && !impStackTop(0).val->IsCnsIntOrI())
             {
-                // byte and sbyte would require more work to support
+                // non-constant byte and sbyte would require more work to support
                 break;
             }
```

Comment:
We shouldn't need this limitation. For shift in particular, working with `constant` or `non-constant` data is the same since we have overloads that take `byte shiftAmount` and overloads that take `Vector128<T> shiftAmount` with the latter simply being `Vector128.CreateScalarUnsafe(shiftAmount)` of the non-constant `byte`.

Reply:
That handling of `constant` vs `non-constant` is general and simply part of the existing `gtNewSimdBinOp` handling, so you shouldn't need to do anything additional really.

We'll already have appropriately masked off the shift amount to be `0-7` and converted it to a `Vector128` if necessary; so you should just need to generate the `shift` followed by the `and` to mask off the upper `n-bits` of each element.

It basically just becomes that if its a constant, you can generate a new icon directly and otherwise the mask is `gtNewSimdBroadcastNode` of `0xFF >> shiftAmount`

Reply:
I updated it now to handle non-const shift amounts as well, but it feels a bit hacky with the temporary `nonConstantByteShiftCountOp` - please let me know if I'm on the right track.

Right now the non-const path is something like
```c#
static Vector128<byte> Shift(Vector128<byte> input, int shiftAmount)
{
    int maskedShiftAmount = shiftAmount & 7;
    Vector128<int> shiftVector = Sse2.ConvertScalarToVector128Int32(maskedShiftAmount);
    Vector128<byte> shiftedInput = Sse2.ShiftRightLogical(input.AsInt32(), shiftVector).AsByte();
    return shiftedInput & Vector128.Create((byte)(255 >> maskedShiftAmount));
}
```